<launch>
  <!-- Total number of robots -->
  <arg name="num_robots" default="1"/>
  <!-- Unique integer id of this robot. -->
  <arg name="robot_id" default="0"/>


  <!-- Launch Kimera Distributed Loop Closure Node -->
  <node name="distributed_loop_closure_node" pkg="kimera_distributed" type="kimera_distributed_loop_closure_node" output="screen" ns="distributed_loop_closure">
    <param name="robot_id"   value="$(arg robot_id)"/>
    <param name="num_robots" value="$(arg num_robots)" />
    <rosparam file="$(find kimera_distributed)/params/visual_loopclosure.yaml"/>
    <param name="vocabulary_path" value="$(find kimera_vio)/vocabulary/ORBvoc.yml"/>
  </node>    

  <!-- Launch Kimera Distributed PCM Node -->
  <node name="distributed_pcm_node" pkg="kimera_distributed" type="kimera_distributed_pcm_node" output="screen" ns="distributed_pcm">
    <param name="robot_id"   value="$(arg robot_id)"/>
    <param name="num_robots" value="$(arg num_robots)" />
    <param name="pcm_threshold_translation" value="0.005" />
    <param name="pcm_threshold_rotation" value="0.005" />
  </node>    
</launch>
